state/in_memory_store.go

```diff
@@ -95,15 +94,15 @@ func (s *InMemoryStore) SetTask(newTaskConf config.TaskConfig) error {
 
 	for ix, taskConf := range *taskConfs {
 		if config.StringVal(taskConf.Name) == newTaskName {
-			// patch update the existing task
-			updatedTaskConf := taskConf.Merge(&newTaskConf)
+			// overwrite the existing task
```

Comment:
I would just add a debug log that this is what happened, only minor suggestion

Reply:
Since the rest of the in-memory store doesn't use logs, I'm not sure we should introduce it. It's relatively simple behavior that's covered by unit tests, so I'm not sure that we need to be that verbose about it.